tests/test_configdata.cpp

```diff
@@ -1144,3 +1144,71 @@ TEST_CASE("config message example 4 - complex conflict resolution", "[config][ex
     CHECK(m_alt1.seqno() == 127);
     CHECK(m_alt1.hash() == m127.hash());
 }
+
+template <typename T, size_t N>
+const std::string_view s_view(const std::array<T, N>& data) {
+    return to_sv(ustring_view{data.data(), data.size()});
+}
+
+inline bt_list lagged_from(MutableConfigMessage m) {
+    return {m.seqno(), s_view(m.hash()), m.diff()};
+}
+
+TEST_CASE("merge test variant 1", "[config][merge]") {
+    MutableConfigMessage m0_x;
```

Comment:
i think if you value initialize `m0_x` here the unit test may pass. looks like it has a bunch of uninitialized memory in the failed test cases (maybe).

Reply:
rather, value initialize all your locals, it is something that is expected usually. i dont know if we assume that we need to value initialize  or not.

Reply:
> i think if you value initialize `m0_x` here the unit test may pass. looks like it has a bunch of uninitialized memory in the failed test cases (maybe).

That won't matter here; the distinction between `T x{};` and `T x;` only exists when `T` has a trivial default constructor with trivial-constructible members (in other words: when it looks like a C struct).